app/utils/date-formatting.js

```diff
@@ -24,8 +24,15 @@ function verifyTimeStamp(dateString) {
 }
 
 // Parses a string to a Moment
-function parseDateString(value) {
-    return value ? moment(verifyTimeStamp(value), parseDateFormats, true) : undefined;
+function parseDateString(value, offset) {
+    // We need the offset here, otherwise the date will be parsed
+    // in UTC timezone
+    moment.tz.setDefault(offset);
+
+    // If the user clears out the date input field, the value will be a moment by default. In that case,
+    // verifyTimeStamp() runs into an error. Therefore, we will just return the value in the parseDateFormats
+    // format.
+    return moment.isMoment(value) ? moment(value, parseDateFormats, true) : moment(verifyTimeStamp(value), parseDateFormats, true);
```

Comment:
> Would you be better off here making sure that post.publishedAtOffset returns null when publishedAt is null?

@kevinansfield did you see the change to that? So you reckon, I should return null instead of the modified (without `verifyTimeStamp()` return value?